azure-pipelines-build.yml

```diff
@@ -0,0 +1,114 @@
+          - template: ../steps/node_script.yml
+            parameters:
+              script: npm ci
+          - template: ../steps/node_script.yml
+            parameters:
+              condition: always()
+              script: npm run tscheck
+          - template: ../steps/node_script.yml
+            parameters:
+              condition: always()
+              script: npm run lint:js
+          - template: ../steps/node_script.yml
+            parameters:
+              condition: always()
+              script: npm run test
+          - template: ../steps/check_changed_files.yml
+            parameters:
+              developmentBranch: develop
+              pathFilters:
+                - name: packages
+                  path: packages
+                - name: web
+                  path: apps/web
+                - name: api
+                  path: apps/api
+              stepName: check_files
+      - name: Build Web
+        condition: >-
+          and(
+            succeeded(),
+            or(
+              and(
+                eq(variables['Build.Reason'], 'Manual'),
+                eq(${{ parameters.buildWeb }}, 'true')
+              ),
+              or(
+                eq(dependencies.Run_Linting_And_Tests.outputs['check_files.packages'], 'true'),
+                eq(dependencies.Run_Linting_And_Tests.outputs['check_files.web'], 'true')
+              )
+            )
+          )
+        dependsOn:
+          - Run Linting & Tests
+        steps:
+          - template: ../steps/azure_web_app_acr_push.yml@templates
+            parameters:
+              azurecrName: $(azurecrName)
+              dockerfilePath: $(dockerFilePath)
+              repository: $(repository)
+              workingDirectory: $(Build.Repository.LocalPath)
+              runTests: false
+        variables:
+          - template: apps/web/pipelines/azure-pipelines-variables.yml@self
+      - name: Build API
+        condition: >-
+          and(
+            succeeded(),
+            or(
+              and(
+                eq(variables['Build.Reason'], 'Manual'),
+                eq(${{ parameters.buildAPI }}, 'true')
+              ),
+              or(
+                eq(dependencies.Run_Linting_And_Tests.outputs['check_files.packages'], 'true'),
+                eq(dependencies.Run_Linting_And_Tests.outputs['check_files.api'], 'true')
+              )
+            )
+          )
+        dependsOn:
+          - Run Linting & Tests
+        steps:
+          - template: ../steps/azure_web_app_acr_push.yml@templates
+            parameters:
+              azurecrName: $(azurecrName)
+              dockerfilePath: $(dockerFilePath)
+              repository: $(repository)
+              workingDirectory: $(Build.Repository.LocalPath)
+              runTests: false
+        variables:
+          - template: apps/api/pipelines/azure-pipelines-variables.yml@self
+    validateName: Lint, Test, & Build Applications
```

Comment:
I think this line has the effect of creating a build task reported as:
Planning-Inspectorate.back-office (Lint, Test, & Build Applications Run Linting & Tests) 

This is a bit of a mouthful – perhaps this `validateName` should be "Prebuild" or something, so we can achieve a shorter (Prebuild - Run Linting & Tests) ?

Reply:
This is the name of the "Stage". And since we only have 1 stage, this is actually completely irrelevant as it doesn't display in the GUI anywhere. In the templates there is a default value of "Validate" so I'll just use that and delete this line. It's only used for referencing within this template (i.e. if one stage depends on another). But like I said, we only have 1 stage (which includes 3 jobs) so the default should be fine.